agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -61,27 +67,70 @@ public static Configuration create(Path agentJarPath) throws IOException {
-            JmxMetric classCountJmxMetric = new JmxMetric();
-            classCountJmxMetric.objectName = "java.lang:type=ClassLoading";
-            classCountJmxMetric.attribute = "LoadedClassCount";
-            classCountJmxMetric.display = "Loaded Class Count";
-            preview.jmxMetrics.add(classCountJmxMetric);
+    private static void loadLogCaptureEnvVar(PreviewConfiguration preview) {
+        final String logCaptureThreshold = overlayWithEnvVar(APPLICATIONINSIGHTS_LOG_CAPTURE_THRESHOLD, "INFO");
+        if (logCaptureThreshold != "INFO") {
+            configurationMessages.add(new ConfigurationMessage("Update default log capture threshold from INFO to {}", logCaptureThreshold));
+            final Map<String, Object> threshold = new HashMap<String, Object>() {{
+                put("threshold", logCaptureThreshold);
+            }};
+            preview.instrumentation = new HashMap<String, Map<String, Object>>() {{
+                put("logging", threshold);
+            }};
         }
+    }
 
-        return config;
+    private static void loadJmxMetrics(PreviewConfiguration preview) throws IOException {
+        String jmxMetricsEnvVarJson = overlayWithEnvVar(APPLICATIONINSIGHTS_JMX_METRICS, null);
+
+        // JmxMetrics env variable has higher precedence over jmxMetrics config from ApplicationInsights.json
+        if (jmxMetricsEnvVarJson != null && !jmxMetricsEnvVarJson.isEmpty()) {
+            configurationMessages.add(new ConfigurationMessage("Update JMX metrics to {}", jmxMetricsEnvVarJson));
+            Moshi moshi = new Moshi.Builder().build();
+            Type listOfJmxMetrics = Types.newParameterizedType(List.class, JmxMetric.class);
+            JsonReader reader = JsonReader.of(new Buffer().writeUtf8(jmxMetricsEnvVarJson));
+            reader.setLenient(true);
+            JsonAdapter<List<JmxMetric>> jsonAdapter = moshi.adapter(listOfJmxMetrics);
+            preview.jmxMetrics = jsonAdapter.fromJson(reader);
+        } else {
+            if (!jmxMetricExisted(preview.jmxMetrics, "java.lang:type=Threading", "ThreadCount")) {
+                JmxMetric threadCountJmxMetric = new JmxMetric();
+                threadCountJmxMetric.objectName = "java.lang:type=Threading";
+                threadCountJmxMetric.attribute = "ThreadCount";
+                threadCountJmxMetric.display = "Current Thread Count";
+                preview.jmxMetrics.add(threadCountJmxMetric);
+            }
+            if (!jmxMetricExisted(preview.jmxMetrics, "java.lang:type=ClassLoading", "LoadedClassCount")) {
+                JmxMetric classCountJmxMetric = new JmxMetric();
+                classCountJmxMetric.objectName = "java.lang:type=ClassLoading";
+                classCountJmxMetric.attribute = "LoadedClassCount";
+                classCountJmxMetric.display = "Loaded Class Count";
+                preview.jmxMetrics.add(classCountJmxMetric);
+            }
```

Comment:
we want to add these additional default metrics whether or not env var was set, similar to how cpu and mem metrics are always captured